x264: bump to build 155

jstebbins opened the pull request:
Requires NASM to build
Unified 8 and 10 bit library support
AVX-512 optimizations
Various other bug fixes and improvements

jstebbins committed:
x264: bump to build 155

Requires NASM to build
Unified 8 and 10 bit library support
AVX-512 optimizations
Various other bug fixes and improvements

bradleysepos commented:
Are we planning to merge this after release (target 1.2.0)?

jstebbins commented:
Whatever you guys want.  I'll still be using the system version of x264 on linux.  Just thought I would get this done since it required more changes than a typical bump.

bradleysepos commented:
As much as I'd like the new features, I think we should let this one soak in the nightly a bit.

bradleysepos commented:
Well, that didn't do what I wanted at all.

bradleysepos commented:
@jstebbins I tried GitHub's resolve conflicts tool and regret it. You may want to do a hard reset to 
9547979. I don't have permissions.

bradleysepos commented:
Oh hey I figured it out. Wheeeeee

bradleysepos commented:
I'm updating `scripts/mac-toolchain-build` to add `nasm`.

What packages are needed on Ubuntu/Fedora? Will need to update the documentation.

jstebbins commented:
A very recent version of nasm is required (2.13).  Fedora 27 and Ubuntu 17.10 have it.  But earlier releases do not.  I think if we want to avoid pain, we should build a local copy for all platforms for the foreseeable future.

But package is called simply "nasm" in both Fedora and Ubuntu.

bradleysepos commented:
Okay. I'm not a fan of the local tools and intend to remove them, just haven't gotten around to it. But that would probably be the least pain.

On Mac, x264 isn't finding nasm despite it being on my path. What a pain.

jstebbins commented:
I'm ok with removing all the local tools.  But it's going to increase the documentation burden. We'll have to have good detailed documentation on how to set up build environments or we'll just be repeating ourselves on the forums all the time to help people through it.

bradleysepos commented:
The documentation is complete regarding dependencies for all systems we support. A script is included to build all dependencies on Mac. Local tools can die whenever we get to it.

bradleysepos commented:
From x264's `config.log`:

```
checking whether nasm supports vmovdqa32 [eax]{k1}{z}, zmm0... no
Failed commandline was:
--------------------------------------------------
nasm conftest.asm  -I. -I$(SRCPATH) -DARCH_X86_64=1 -I$(SRCPATH)/common/x86/ -f macho64 -DPIC -DPREFIX  -o conftest.o
nasm: fatal: unrecognised output format `macho64' - use -hf for a list
type `nasm -h' for help
--------------------------------------------------
Failed program was:
--------------------------------------------------
vmovdqa32 [eax]{k1}{z}, zmm0
--------------------------------------------------
```

```
$ nasm -hf | grep macho64
    macho64   NeXTstep/OpenStep/Rhapsody/Darwin/MacOS X (x86_64) object files
```

:frowning_face:

jstebbins commented:
nasm --version?

bradleysepos commented:
```
$ nasm --version
NASM version 2.13.02 compiled on Dec 28 2017
```

bradleysepos commented:
I think it's ignoring `PATH` and picking up macOS' `/usr/bin/nasm`. 🙃

bradleysepos committed:
contrib: Remove yasm from x264 module.

bradleysepos committed:
scripts: Update mac-toolchain-build to add nasm.

bradleysepos committed:
build: Update configure.py to check for nasm >= 2.13.

Required for recent x264. Better to fail here than part-way through a build.

bradleysepos commented:
I added some supporting commits but still have no idea why x264 is not picking up `nasm` on my system.

bradleysepos commented:
Looks like it actually works when cross-compiling, though I also get `Unknown option --build=x86_64-apple-darwin17.3.0, ignored`. The only other things that should be different are `--host` and `--cross-prefix`. ~~~I wonder whether setting `--host` for a non-cross build will work.~~~ Nope.

bradleysepos committed:
contrib: Make x264 cross options valid for all cross, not just mingw.

Also set CONFIGURE.build to null since the configure script has no clue about --build.

bradleysepos commented:
The only way I can seem to make this work is to hard code `AS="/usr/local/bin/nasm"` (or wherever nasm >= 2.13 is installed) in `configure`. Something's amiss.

bradleysepos committed:
scripts: Enable sections and lto for nasm.

Pass a couple standard variables since make sometimes makes zero sense.

bradleysepos committed:
build: Unclobber PATH when shunting build through Xcode.

Tools shipping with Xcode are still in PATH, only moved from first to last priority. Avoids accidentally using old tools where a newer version is installed. This ensures the tools identified by configure will be the exact tools used by make via Xcode.

bradleysepos commented:
eadfc60 should fix the old nasm problem on Mac.

jstebbins committed:
encx264: check correct build number in ifdef

The multi-lib change actually happend in build 153

kodawah commented:
~shouldn't you use `X264_BUILD 153` to check for the new bitdepth field?~
never mind I missed the last patch in the series

bradleysepos commented:
I've tested with this enough that I think it's probably safe for 1.1.0 if we want to include it.

bradleysepos commented:
If nasm in distros is a sticking issue, we can push it back again.

bradleysepos commented:
Already have one report of HandBrake not compiling with newer system x264, so perhaps we should get this in. Can add nasm compilation notes to the developer docs.

bradleysepos commented:
Merged as we need the libhb changes to support newer distros now. Will add nasm instructions to the docs shortly.
